src/text.rs

```diff
@@ -238,6 +239,17 @@ impl<'a> From<&'a str> for Span<'a> {
     }
 }
 
+impl<'a> Styled for Span<'a> {
+    type Item = Span<'a>;
+    fn style(&self) -> Style {
+        self.style
+    }
+
+    fn set_style(self, style: Style) -> Self {
+        Self::styled(self.content, style)
+    }
+}
+
```

Comment:
Would making a `style_mut()` method instead work for this? It's a pattern that I've seen a lot, and might simplify the code below to something like:
```rust
self.style_mut().fg(color.into());
self
```

Reply:
I'm not sure how to implement that

Reply:
```rust
fn style_mut(&mut self) -> &mut Style {
  &mut self.style;
}
```

Reply:
I was talking about the trait, sorry for not being clear.

Reply:
The `style_mut()` change is one that I'm not 100% sure is possible either (it's probably some sort of funky combination of defining `From<...> for Styled` and then `Stylized` taking an `Into<Styled>` instead. Not sure it's a big deal though.

Reply:
I'm not sure if having a reference only to the style works. Isn't it necessary to also have a reference to the widget (`Self::Item`)? (for the `Styled` trait)